src/App.tsx

```diff
@@ -1,39 +1,48 @@
 import React from 'react';
 import './App.scss';
+import { Clock } from './components/Clock/Clock';
 
 function getRandomName(): string {
   const value = Math.random().toString().slice(2, 6);
 
   return `Clock-${value}`;
 }
 
-export const App: React.FC = () => {
-  const date = new Date();
-  const clockName = getRandomName();
+type State = {
+  hasClock: boolean;
+  clockName: string;
+};
+
+export class App extends React.Component<{}, State> {
+  state = {
+    hasClock: true,
+    clockName: getRandomName(),
+  };
 
-  // This code starts a timer
-  const timerId = window.setInterval(() => {
-    // ...
-  }, 1000);
+  componentDidMount() {
+    document.addEventListener('contextmenu', () => {
+      this.setState({ hasClock: false });
+    });
+    document.addEventListener('click', () => {
+      this.setState({ hasClock: true });
+    });
 
-  // this code stops the timer
-  window.clearInterval(timerId);
+    window.setInterval(() => {
+      const clockName = getRandomName();
 
-  return (
-    <div className="App">
-      <h1>React clock</h1>
+      this.setState({ clockName });
+    }, 3300);
+  }
```

Comment:
I've found an issue here: when your component unmounts this function keeps working in background. You need to clearInterval() before it will unmount.
[React-clock.webm](https://user-images.githubusercontent.com/54940595/187459991-4135fe4e-985d-4296-89da-52dcd1cad153.webm)